src/main/java/org/skriptlang/skript/bukkit/damagesource/elements/ExprDirectEntity.java

```diff
@@ -0,0 +1,88 @@
+package org.skriptlang.skript.bukkit.damagesource.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.damage.DamageSource;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.damagesource.DamageSourceExperiment;
+import org.skriptlang.skript.bukkit.damagesource.elements.ExprSecDamageSource.DamageSourceSectionEvent;
+
+@Name("Damage Source - Direct Entity")
+@Description({
+	"The direct entity of a damage source.",
+	"The direct entity is the entity that directly caused the damage. (e.g. the arrow that was shot)",
+	"Attributes of a damage source cannot be changed once created, only while within the 'custom damage source' section."
+})
+@Example("""
+	set {_source} to a custom damage source:
+		set the damage type to magic
+		set the causing entity to {_player}
+		set the direct entity to {_arrow}
+		set the damage location to location(0, 0, 10)
+	damage all players by 5 using {_source}
+	""")
+@Example("""
+	on death:
+		set {_direct} to the direct entity of event-damage source
+	""")
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.20.4+")
+public class ExprDirectEntity extends SimplePropertyExpression<DamageSource, Entity> implements DamageSourceExperiment {
+
+	static {
+		registerDefault(ExprDirectEntity.class, Entity.class, "direct entity", "damagesources");
```

Comment:
i was thinking about the direct/indirect syntaxes and had a possible suggestion
Since `direct entity of ...` doesn't really mean much on its own, what do you think about:
```
[in]direct (source|cause) of %damagesource%
```
I'm not sure about source of source or whatever, and maybe it'd be better to include the work 'entity'.
You can also combine this class and the indirect class

idk it might be better it might be worse, let me know what you think

Reply:
What do you think about the syntax change, though?